src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PrivateDataRetrievalService.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.companieshouse.overseasentitiesapi.utils;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import uk.gov.companieshouse.api.model.beneficialowner.PrivateBoDataListApi;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+@Component
+public class PrivateDataRetrievalService {
+
+    public static final String COMPANY_NUMBER = "company_number";
+
+    private final ApiClientService apiClientService;
+
+    public PrivateDataRetrievalService(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+
+    public void initialisePrivateData(String companyNumber)
+            throws ServiceException {
+        getBeneficialOwnersData(companyNumber);
+    }
+
+    private PrivateBoDataListApi getBeneficialOwnersData(String companyNumber)
+            throws ServiceException {
+
+        var logMap = new HashMap<String, Object>();
+        logMap.put(COMPANY_NUMBER, companyNumber);
+        ApiLogger.info("Retrieving Beneficial Owners for Company Number " + logMap);
+
+        try {
+            return apiClientService.getInternalApiClient()
+                    .privateBeneficialOwnerResourceHandler()
+                    .getBeneficialOwner("/overseas-entity/" + companyNumber + "/beneficial-owners")
+                    .execute()
+                    .getData();
```

Comment:
Maybe log out the count of officers you found - that could be useful info for testing.